### PR TITLE
Add navigation opt out modal

### LINKED ISSUE #5479: New navigation - getting feedback on opt-out
**User story:**
As a merchant
I want to have a way to provide feedback about the new navigation
When I opt out of the navigation in Settings

**Description:** When a user opts out of the new navigation from **Settings > Advanced > Features**, a modal will show with a link to Share Feedback in a survey. 

**Design:** Upon de-selecting the navigation option and saving the new settings, a modal will show.

Modal content (copy is WIP):
Header: Help us improve
Body: Take this 2-minute survey to share why you're opting out of the new navigation
[No thanks][Share feedback]

Choosing `No thanks` will close the modal.
Choosing `Share feedback` will link to the Crowdsignal survey: [https://automattic.survey.fm/new-navigation-opt-out](https://automattic.survey.fm/new-navigation-opt-out)

Acceptance criteria:
[] TBD
Event tracking:
TBD

cc @pmcpinto @joshuatf

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #5479

Adds a modal to request feedback after opting out of the navigation experience.

### Screenshots

Screen Shot 2020-12-09 at 4 27 32 PM

### Detailed test instructions:

1. Enable the nav feature at `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
1. Make sure no modal is shown.
1. Disable the nav feature.
1. Make sure the modal is shown.
1. Check that "Share Feedback" opens the link as expected and the other actions (including clicking "Share Feedback") close the modal.
1. Check that the modal is not persisted upon refreshing the page or navigating to another page.

### CODE REVIEW

**Review by jeffstieler — COMMENTED** [review]
This PR looks good, but I noticed during testing that the modal gets displayed every time you disabled the navigation feature, not just the first time. Is that expected?

**Comment by joshuatf** [comment]
@jeffstieler I wasn't sure on whether or not this was expected or not either, but opted to show it on each disable.  /cc @pmcpinto @elizaan36 

My reasoning being that it seems feasible that the navigation was turned back on after some period of time, but still fell short of the store owner's expectations.  That new feedback may be as equally valuable as the original.  It's also fairly non-intrusive and easy to close by clicking outside the modal.

**Comment by elizaan36** [comment]
> My reasoning being that it seems feasible that the navigation was turned back on after some period of time, but still fell short of the store owner's expectations

@joshuatf I think that makes a lot of sense. it would be interesting to know why someone re-enabled then re-disabled.

**Review by jeffstieler — APPROVED** [review]
👍

File: client/wp-admin-scripts/navigation-opt-out/container.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { Button, Modal } from '@wordpress/components';
+
+export default class NavigationOptOutContainer extends Component {
```
Comment (jeffstieler) [inline client/wp-admin-scripts/navigation-opt-out/container.js:8]: @samueljseay won't like this 😛

Comment (joshuatf) [reply client/wp-admin-scripts/navigation-opt-out/container.js]: No longer exporting as default - happy holidays @samueljseay!

**Pull request merged**